src/tests/Loader/ContextualReflection/ContextualReflection.cs

```diff
@@ -730,10 +730,11 @@ void TestDefineDynamicAssembly(bool collectibleContext, AssemblyBuilderAccess as
         {
             AssemblyLoadContext assemblyLoadContext = collectibleContext ? new AssemblyLoadContext("DynamicAssembly Collectable context", true) : AssemblyLoadContext.Default;
             AssemblyName dynamicAssemblyName = new AssemblyName("DynamicAssembly");
+            AssemblyBuilder assemblyBuilder;
 
             using (assemblyLoadContext.EnterContextualReflection())
             {
-                AssemblyBuilder assemblyBuilder = AssemblyBuilder.DefineDynamicAssembly(dynamicAssemblyName, assemblyBuilderAccess);
+                assemblyBuilder = AssemblyBuilder.DefineDynamicAssembly(dynamicAssemblyName, assemblyBuilderAccess);
```

Comment:
I don't understand the change... what is this fixing?

Reply:
The `AssemblyBuilder` was in the local scope of the `using` statement. If a dynamic assembly is created with the `AssemblyBuilderAccess.RunAndCollect` parameter, this could lead to the fact that at the time of checking the `Assert` condition it was already removed from memory. I think changing the scope of the `AsemblyBuilder` instance will solve the problem of floating test failure.

Reply:
`assemblyBuilder` is never used; in a release build, these should produce exactly the same IL.  Even if this fixes it for a debug build, where this could extend the lifetime of the instance until after the using block and prevent it from being collected before then, is this test never run in a release build?

Reply:
Yes, the IL code was the same. Rewrote the test so that the AssemblyBuilder instance is used in the Assert test. The local load test did not issue any errors for an hour.